docs/reference/index-modules.asciidoc

```diff
@@ -82,6 +82,39 @@ breaking change].
     after segments are merged. Segment merging can be forced using
     <<indices-forcemerge,force merge>>.
 
+[[index-mode-setting]] `index.mode`::
+
+    preview::["The `logs` index mode is in technical preview and may be changed or removed in a future release. Elastic will work to
+    fix any issues, but features in technical preview are not subject to the support SLA of official GA features."]
+
+    The `index.mode` setting is used to control settings applied in specific domains like ingestion of time series data or logs.
+    Different mutually exclusive modes exist, which are used to apply settings or default values controlling indexing of documents,
+    sorting and other parameters whose value affects indexing or query performances.
+
+    [source,console]
+    --------------------------------------------------
+    PUT my-index-000001
+    {
+      "settings": {
+        "index": {
+          "mode": "standard" <1>
+        }
+      }
+    }
+    --------------------------------------------------
+
+    <1> This index uses the `standard` index mode
+
+    Index mode supports the following values:
+
+    `null`::: Default value (same as `standard`)
+
+    `standard`::: Standard indexing with default settings.
+
+    `time_series`::: Controls time series indexing and is documented in <<tsds-index-settings,TSDS Settings>>
+
+    `logs`::: Controls indexing of logs. It applies default sort settings on the `hostname` and `timestamp` fields (tech preview feature).
```

Comment:
```suggestion
[[index-mode-setting]] `index.mode`::
+
The `index.mode` setting is used to control settings applied in specific domains like ingestion of time series data or logs.
Different mutually exclusive modes exist, which are used to apply settings or default values controlling indexing of documents,
sorting and other parameters whose value affects indexing or query performances.
+
[source,console]
----------------
PUT my-index-000001{
   "settings":{
      "index":{
         "mode":"standard" <1>
      }
   }
----------------
+
<1> This index uses the `standard` index mode
+
Index mode supports the following values:

`null`::: Default value (same as `standard`)

`standard`::: Standard indexing with default settings.

`time_series`::: Controls time series indexing and is documented in <<tsds-index-settings,TSDS Settings>>

`logs`::: Controls indexing of logs. It applies default sort settings on the `hostname` and `timestamp` fields.
preview:[]
```

Reply:
@salvatore-campagna you've hit some gnarly asciidoc syntax stuff here alright, I tried this approach locally and it seems to work :)